component/pyroscope/scrape/target.go

```diff
@@ -400,3 +400,27 @@ func targetsFromGroup(group *targetgroup.Group, cfg Arguments) ([]*Target, []*Ta
 
 	return targets, droppedTargets, nil
 }
+
+func inferServiceName(lset labels.Labels) string {
+	if k8sNamespace := lset.Get("__meta_kubernetes_namespace"); k8sNamespace != "" {
+		k8sNameLabels := []string{
+			"__meta_kubernetes_pod_label_pyroscope_name",
+			"__meta_kubernetes_pod_label_app_kubernetes_io_name",
+			"__meta_kubernetes_pod_container_name",
+			"__meta_kubernetes_pod_name",
+		}
```

Comment:
Should we make agent to understand `pyroscope.io/application-name` annotation (see [example](https://github.com/grafana/pyroscope/tree/main/examples/golang-pull/kubernetes)) for backward compatibility with the existing deployments profiled with pyroscope OSS?

A nit: we could allocate the slice globally

Reply:
I think if we try to support pyroscope.io annotations, we should support all of them, not just one and supporting all of the may complicate logic

Reply:
lets definitely not support all of them. I think if we support any we should support only the applicaiton name, but I'm also fine not supporting this and when we release pyroscope 1.0 we can have people make the change

Reply:
Plus not sure about using the `pyroscope.io` prefix.

Reply:
My idea is to make the configuration process less prone to human errors. I expect that a good share of users that deal with this configuration already have experience of configuring Pyroscope OSS, where we have the `application-name` annotation. I anticipate that due to the human factor, these two might be mixed mistakenly.